app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java

```diff
@@ -34,9 +34,9 @@ public void onButtonParkingSpaceReservationPickerPressed(DatePickerDialog.OnDate
     @Override
     public void onButtonParkingSpaceReservationSavePressed() {
         model.completeReservationInfo(view.getParkingSpace(), view.getSecurityCode());
-        ReservationVerificationResult reservationVerificationResult = model.checkFields(model.getReservation());
+        ReservationVerificationResult reservationVerificationResult = model.checkFields();
         if (reservationVerificationResult == ReservationVerificationResult.SUCCESS) {
-            model.makeReservation();
+            model.getDatabase().addReservation(model.getReservation());
```

Comment:
the model should not return the database. please call a method into the model to do this

Reply:
Done!